src/views/dashboard/DashboardIndex.vue

```diff
@@ -1,112 +1,391 @@
+import { showSuccess, showError } from '@nextcloud/dialogs'
 
 export default {
 	name: 'DashboardIndex',
 	components: {
 		NcAppContent,
 		NcEmptyContent,
+		NcLoadingIcon,
+		NcActions,
+		NcActionButton,
 		DatabaseOutline,
 		FileCodeOutline,
+		ChevronDown,
+		ChevronUp,
+		DotsHorizontal,
+		Pencil,
+		TrashCanOutline,
+		Upload,
+		Export,
+		ApiIcon,
+		Download,
+		Calculator,
 	},
-	data() {
+	setup() {
+		const store = useDashboardStore()
+		const navigationStore = useNavigationStore()
+		const expandedSchemas = ref(new Set())
+		const calculating = ref(null)
+
+		// Computed properties
+		const loading = computed(() => store.loading)
+		const error = computed(() => store.error)
+		const registers = computed(() => store.registers)
+
+		// Methods
+		const toggleSchema = (schemaId) => {
+			if (expandedSchemas.value.has(schemaId)) {
+				expandedSchemas.value.delete(schemaId)
+			} else {
+				expandedSchemas.value.add(schemaId)
+			}
+		}
+
+		const formatBytes = (bytes) => {
+			if (!bytes || bytes === 0) return '0 KB'
+			const k = 1024
+			const sizes = ['B', 'KB', 'MB', 'GB']
+			const i = Math.floor(Math.log(bytes) / Math.log(k))
+			return `${parseFloat((bytes / Math.pow(k, i)).toFixed(2))} ${sizes[i]}`
+		}
+
+		const calculateSizes = async (register) => {
+			calculating.value = register.id
+			try {
+				await axios.post(`/index.php/apps/openregister/api/dashboard/calculate/${register.id}`)
+				showSuccess(t('openregister', 'Sizes calculated successfully'))
+				await store.fetchRegisters() // Refresh data
+			} catch (error) {
+				showError(t('openregister', 'Failed to calculate sizes'))
+				console.error('Failed to calculate sizes:', error)
+			} finally {
+				calculating.value = null
+			}
+		}
+
+		const downloadOas = async (register) => {
+			const baseUrl = window.location.origin
+			const apiUrl = `${baseUrl}/index.php/apps/openregister/api/registers/${register.id}/oas`
+			try {
+				const response = await axios.get(apiUrl)
+				const blob = new Blob([JSON.stringify(response.data, null, 2)], { type: 'application/json' })
+				const downloadLink = document.createElement('a')
+				downloadLink.href = URL.createObjectURL(blob)
+				downloadLink.download = `${register.title.toLowerCase()}-api-specification.json`
+				document.body.appendChild(downloadLink)
+				downloadLink.click()
+				document.body.removeChild(downloadLink)
+				URL.revokeObjectURL(downloadLink.href)
+			} catch (error) {
+				showError(t('openregister', 'Failed to download API specification'))
+				console.error('Error downloading OAS:', error)
+			}
+		}
+
+		const viewOasDoc = (register) => {
+			const baseUrl = window.location.origin
+			const apiUrl = `${baseUrl}/index.php/apps/openregister/api/registers/${register.id}/oas`
+			window.open(`https://redocly.github.io/redoc/?url=${encodeURIComponent(apiUrl)}`, '_blank')
+		}
+
+		// Fetch data on component mount
+		onMounted(() => {
+			store.fetchRegisters()
+		})
```

Comment:
We alsways set functions in the methods:
~~~
methods: {
		async loadRegisters() {
			await registerStore.refreshRegisterList()
			this.registers = registerStore.registerList
		},
	},
	~~~
And mounted in mounted too

Reply:
Changed